tests/components/zone/test_init.py

```diff
@@ -199,6 +338,127 @@ async def test_active_zone_prefers_smaller_zone_if_same_distance_2(hass):
     assert active.entity_id == "zone.smallest_zone"
 
 
+async def test_zone_smaller_than_accuracy(hass):
+    """Test zone size preferences."""
+    latitude = 32.880600
+    longitude = -117.237561
+    assert await setup.async_setup_component(
+        hass,
+        zone.DOMAIN,
+        {
+            "zone": [
+                {
+                    "name": "Smallest Zone",
+                    "latitude": latitude,
+                    "longitude": longitude,
+                    "radius": 50,
+                }
+            ]
+        },
+    )
+
+    # Should not enter zone, accuracy is poor
+    active = zone.async_active_zone(hass, latitude, longitude, 51)
+    assert active is None
+
+    # Should enter zone, accuracy is good enough
+    active = zone.async_active_zone(hass, latitude, longitude, 49, active)
+    assert active.entity_id == "zone.smallest_zone"
+
+    # Should remain in zone
+    active = zone.async_active_zone(hass, latitude, longitude, 51, active)
+    assert active.entity_id == "zone.smallest_zone"
+
+
+def assert_zone(zone, expected_zone, idx):
+    """Assert current zone is the expected one.
+
+    Unused parameter idx added to get useful debug information from pytest.
+    """
+    if expected_zone is None:
+        assert zone is None
+    else:
+        assert zone.entity_id == f"zone.{expected_zone}"
+
+
+@pytest.mark.parametrize(
+    "zone_radii,start_location,locations,expected_zones",
+    # zone radii: radius for each configured zone
+    # start_location: initial latitude, longitude, accuracy
+    # locations: list of (distance, accuracy) from starting location
+    # expected zones: list of expected zone for each location in the locations list
+    [
+        # Don't enter a zone smaller than the location's accuracy
+        (
+            (6, 12),
+            (55.69991353346531, 13.206717073911985, 6),
+            [(0, 12), (0, 11.9), (0, 3)],
+            [None, "kebab_shop_garden", "kebab_shop"],
+        ),
+        # Don't leave a zone if location with accuracy intersects with zone
+        (
+            (6, 12),
+            (55.69991353346531, 13.206717073911985, 6),
+            [(0, 3), (0, 11.9), (0, 12)],
+            ["kebab_shop", "kebab_shop", "kebab_shop"],
+        ),
+        # Exit a zone if location with accuracy does not intersect with zone
+        (
+            (6, 11),
+            (55.69991353346531, 13.206717073911985, 6),
+            [(0, 3), (6, 3), (9, 3), (6, 3)],
+            ["kebab_shop", "kebab_shop", None, "kebab_shop_garden"],
+        ),
+        # Exit a zone if location with accuracy does not intersect with zone
+        (
+            (6, 13),
+            (55.69991353346531, 13.206717073911985, 6),
+            [(0, 3), (6, 3), (9, 3), (6, 3)],
+            ["kebab_shop", "kebab_shop", "kebab_shop_garden", "kebab_shop_garden"],
+        ),
```

Comment:
These two have the same comment, but the zone radii are slightly different. Should we clarify in the comment what we want to test?